### ISSUE TITLE
Contextual attention flow behaves weirdly on custom dataset

### ISSUE THREAD

**Issue by nsalminen** [JiahuiYu/generative_inpainting#347]
I am currently training an unmodified version of your architecture with a custom dataset. During training, I'm seeing unexpected output and an attention map where the unknown region is white, rather than the known region.

![individualImage4](https://user-images.githubusercontent.com/1269004/68543228-a4141e80-03b4-11ea-9999-12dd248ad305.png)
_TensorBoard image output at step 86,000 with batch size 16_

Is this normal behavior? To try to find out whether this is due to one of my own changes I have reverted back to your original DeepFill V2 version, but still am getting the same results.

I'd appreciate your opinion on this! Thank you!

**Comment by JiahuiYu** [comment]
It seems like this model has already failed. In your dataset, the majority part is empty. One suggestion is to increase the pixel-wise L1 loss to stabilize training.

If you have any further question, please leave a message here and I am happy to help. :)

**Comment by nsalminen** [comment]
Thanks a lot for your suggestion! I will give it a try

**Comment by nsalminen** [comment]
> It seems like this model has already failed. In your dataset, the majority part is empty. One suggestion is to increase the pixel-wise L1 loss to stabilize training.
> 
> If you have any further question, please leave a message here and I am happy to help. :)

@JiahuiYu thanks again for your help! I have tweaked both the loss and my dataset to stabilize training. This is the result I am currently getting:

![rgb_356000](https://user-images.githubusercontent.com/1269004/69224765-6a45d380-0b7d-11ea-897f-f2829cd83fd6.png)

Although the results look very good and I am familiar with optical flow, the attention map confuses me somewhat. I am unsure how to interpret it as it is still behaving in a way that I wouldn't expect it to and I wonder if it's maybe because my dataset contains a lot of empty area.

Do you have an idea what could be going on? Thank you!

**Comment by JiahuiYu** [comment]
#299. Hi the color yellow means this part is copied from the top part (yellow in the attention map is at the top).

**Comment by nsalminen** [comment]
@JiahuiYu When I output the coarse result of the above model I observe that it is always fully blank, while the refined result looks good. Like you mentioned, I think this is a result of the majority of the image being empty.

I have tried tweaking the GAN loss, l1 loss and learning rate, but have not been able to reach a stable way of avoiding this problem in this case. Do you have any suggestions regarding this? I would really appreciate your thoughts when you have time. Thank you!
